new_client/src/components/Header.js

```diff
@@ -1,21 +1,105 @@
-import React from 'react';
-import { Link } from 'react-router-dom';
+import React, { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
 
+import "./Header.css";
+export default function Header(props) {
+	const [isDesktop, setisDesktop] = useState(window.innerWidth > 1080);
+	const [navOpen, setnavOpen] = useState(false);
+	useEffect(() => {
+		function navToggle() {
+			setnavOpen((navPrevState) => {
+				return !navPrevState;
+			});
+		}
+		window.addEventListener("resize", () => {
+			setisDesktop(window.innerWidth > 1080);
+		});
+		document.getElementById("nav-toggler").addEventListener("click", navToggle);
+	}, []);
 
-import './Header.css';
-export default function Header() {
-  return (
-    <div className="navbar">
-      <div className="navbar-right">
-        <Link to="/"><i className="fa fa-fw fa-home"></i> HOME</Link>
-        <Link to="/about"><i className="fa fa-fw fa-info"></i> ABOUT US</Link>
-
-        <Link to="/events"><i className="fa fa-fw fa-calendar"></i>EVENTS</Link>
-        <Link to="/glimpse"><i className="fa fa-fw fa-info"></i>GLIMPSES</Link>
-        <Link to="/blogs"><i className="fa fa-fw fa-pencil"></i>BLOGS</Link>
-        <Link to="/signup"><i className="fa fa-fw fa-sign-in"></i> SIGN UP</Link>
-
-      </div>
-    </div>
-  );
+	return (
+		<div className={isDesktop ? "navbar-custom-desk" : "navbar-custom-mob"}>
+			<div
+				className={isDesktop ? "nav-burg-desk" : "nav-burg-mob"}
+				id="nav-toggler"
+			>
+				<i className="fa fa-bars"></i>
+			</div>
+			<div
+				className={
+					(isDesktop ? "nav-item-list-desk" : "nav-item-list-mob") +
+					(navOpen ? " nav-mob-open" : " nav-mob-close")
+				}
+			>
+				<Link
+					to="/"
+					className={
+						(isDesktop ? "nav-item-desk" : "nav-item-mob") + " curr-page"
```

Comment:
By appending `curr-page`, we imply that this is the current page, if I am not wrong. So the logic should be, whenever we click on the route, `curr-page` should be appended to the already existing classNames. The code currently highlights the Home route, which is correct, but not accurate.